src/wp-includes/class-wpdb.php

```diff
@@ -1765,7 +1791,13 @@ public function prepare( $query, ...$args ) {
 		$args_escaped = array();
 
 		foreach ( $args as $i => $value ) {
-			if ( in_array( $i, $arg_identifiers, true ) ) {
+			if ( in_array( $i, $arg_variadics, true ) ) {
```

Comment:
Instead of `in_array` could we use `isset`?

Reply:
I'm currently using three lists of identifiers/variadics/strings, where the keys aren't being used (so `isset` won't work there)... that said, it might be possible change how this bit works, and instead have a single array, where the key is the argnum, and the value is the type; but I'm not sure about my overall direction yet, and I might go in a [completely different direction](https://github.com/WordPress/wordpress-develop/pull/4990#issuecomment-1680710157).